### PR TITLE
[sdformat12] new port

### PR DESCRIPTION
**Pull request by talregev**
**Describe the pull request**
Add sdformat12 port for gazebo ignition

- #### What does your PR fix?
  Fixes #...

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
 Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

Help needed to recognize `libignition-math6-dev`

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@talregev, Thanks for your pr, please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/79136/artifacts?artifactName=failure%20logs%20for%20x86-windows&api-version=7.0&%24format=zip). Please ping me if you need any help.

**Comment by talregev** [comment]
> @talregev, Thanks for your pr, please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/79136/artifacts?artifactName=failure%20logs%20for%20x86-windows&api-version=7.0&%24format=zip). Please ping me if you need any help.
> 

@JonLiu1993 
I know it does not recognize ignition-math6 and I don't know how to solve it.
I will be happy to help. Thank you!

**Comment by JonLiu1993** [comment]
Ping @talregev  for response.

**Comment by talregev** [comment]
> > @talregev, Thanks for your pr, please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/79136/artifacts?artifactName=failure%20logs%20for%20x86-windows&api-version=7.0&%24format=zip). Please ping me if you need any help.
> 
> @JonLiu1993 I know it does not recognize ignition-math6 and I don't know how to solve it. I will be happy to help. Thank you!

@JonLiu1993 I responded here.

**Comment by JonLiu1993** [comment]
@dg0yt , could you please help take view this error?

File: ports/sdformat12/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO gazebosim/sdformat
+    REF sdformat12_12.0.0
+    SHA512 234e1378bf82108a5d8e8e3e199d554911531ffd36e9610720b0e2ccbd044c185aca1937dcd171c979fd9332dc4f4685c6632f46818a9dec8ff50444b99ba0de
+    HEAD_REF sdf12
+    PATCHES
+        fix-quote.patch
```
Comment (dg0yt) [inline ports/sdformat12/portfile.cmake:9]: This patch isn't needed.

File: ports/sdformat12/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        no-absolute.patch
+)
+
+# Ruby is required by the sdformat build process
+vcpkg_find_acquire_program(RUBY)
+get_filename_component(RUBY_PATH ${RUBY} DIRECTORY)
+set(_path $ENV{PATH})
```
Comment (dg0yt) [inline ports/sdformat12/portfile.cmake:16]: This needs quotes.

File: ports/sdformat12/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    OPTIONS 
+        -DBUILD_TESTING=OFF
+        -DUSE_EXTERNAL_URDF=ON
+        -DUSE_EXTERNAL_TINYXML=ON
+)
+
+vcpkg_cmake_install()
+
+# Restore original path
+set(ENV{PATH} ${_path})
```
Comment (dg0yt) [inline ports/sdformat12/portfile.cmake:30]: This needs quotes.

File: ports/sdformat12/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/sdformat10")
+vcpkg_fixup_pkgconfig()
+
+# Remove debug files
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include"
+                    "${CURRENT_PACKAGES_DIR}/debug/lib/cmake"
+                    "${CURRENT_PACKAGES_DIR}/debug/share")
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (dg0yt) [inline ports/sdformat12/portfile.cmake:42]: This needs quotes. (And may use `maybe vcpkg_install_copyright`.)

**Review by dg0yt — CHANGES_REQUESTED** [review]
I can look at the CI logs once there is current CI build.

**Comment by JonLiu1993** [comment]
> I can look at the CI logs once there is current CI build.

I merged the master, and ci started running again

File: ports/sdformat12/portfile.cmake (hunk 1 above)
Comment (talregev) [reply ports/sdformat12/portfile.cmake]: done.

File: ports/sdformat12/portfile.cmake (hunk 2 above)
Comment (talregev) [reply ports/sdformat12/portfile.cmake]: done.

File: ports/sdformat12/portfile.cmake (hunk 3 above)
Comment (talregev) [reply ports/sdformat12/portfile.cmake]: done.

File: ports/sdformat12/portfile.cmake (hunk 4 above)
Comment (talregev) [reply ports/sdformat12/portfile.cmake]: done.

**Comment by talregev** [comment]
@dg0yt from your comment:
https://github.com/microsoft/vcpkg/pull/28495#issuecomment-1363622738

Thank you for you time of review my PR. I push a fix according to your comments.
But it not solve the main issue that it not recognize the  `ignition-math6` library. 
I will happy to assist or any help to fix this PR. Thank you!

File: ports/sdformat12/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    HEAD_REF sdf12
+    PATCHES
+        no-absolute.patch
+)
+
+# Ruby is required by the sdformat build process
+vcpkg_find_acquire_program(RUBY)
+get_filename_component(RUBY_PATH ${RUBY} DIRECTORY)
+set(_path "$ENV{PATH}")
+vcpkg_add_to_path(${RUBY_PATH})
```
Comment (dg0yt) [inline ports/sdformat12/portfile.cmake:16]: Don't change `PATH`. Pass `"-DRUBY=${RUBY}"` via `OPTIONS` instead.

File: ports/sdformat12/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        -DUSE_EXTERNAL_TINYXML=ON
+)
+
+vcpkg_cmake_install()
+
+# Restore original path
+set(ENV{PATH} "${_path}")
+
+# Fix cmake targets and pkg-config file location
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/sdformat10")
```
Comment (dg0yt) [inline ports/sdformat12/portfile.cmake:32]: `sdformat12` ?

File: ports/sdformat12/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+vcpkg_fixup_pkgconfig()
+
+# Remove debug files
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include"
+                    "${CURRENT_PACKAGES_DIR}/debug/lib/cmake"
```
Comment (dg0yt) [inline ports/sdformat12/portfile.cmake:37]: Isn't this auto-cleaned by `vcpkg_cmake_config_fixup`?

**Review by dg0yt — CHANGES_REQUESTED** [review]
sdformat12 wants ignition-math 6.8, but vcpkg offers only 6.6.

File: ports/sdformat12/portfile.cmake (hunk 7 above)
Comment (talregev) [reply ports/sdformat12/portfile.cmake:37]: I just copy from sdformat10

Comment (dg0yt) [reply ports/sdformat12/portfile.cmake:37]: You can do more than that ;-)

File: ports/sdformat12/portfile.cmake (hunk 5 above)
Comment (talregev) [reply ports/sdformat12/portfile.cmake]: done.

File: ports/sdformat12/portfile.cmake (hunk 6 above)
Comment (talregev) [reply ports/sdformat12/portfile.cmake]: done.

File: ports/sdformat12/portfile.cmake (hunk 7 above)
Comment (talregev) [reply ports/sdformat12/portfile.cmake:35]: done.

**Comment by talregev** [comment]
> sdformat12 wants ignition-math 6.8, but vcpkg offers only 6.6.

Can you update `ignition-math` port to 6.8?

File: ports/sdformat12/portfile.cmake (hunk 7 above)
Comment (talregev) [reply ports/sdformat12/portfile.cmake:35]: Should I delete all of these lines?

**Comment by talregev** [comment]
@JonLiu1993 Can you check the CI?

File: ports/sdformat12/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+# Ruby is required by the sdformat build process
+vcpkg_find_acquire_program(RUBY)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS 
+        -DBUILD_TESTING=OFF
+        -DUSE_EXTERNAL_URDF=ON
+        -DUSE_EXTERNAL_TINYXML=ON
+        -DRUBY=${RUBY}
```
Comment (dg0yt) [inline ports/sdformat12/portfile.cmake:21]: `"-DRUBY=${RUBY}"`

Comment (talregev) [reply ports/sdformat12/portfile.cmake]: done

**Comment by talregev** [comment]
@dg0yt @JonLiu1993 
It ready for review.

File: ports/sdformat12/vcpkg.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "sdformat12",
+  "version": "12.0.0",
+  "description": "Simulation Description Format (SDF) parser and description files.",
+  "homepage": "http://sdformat.org/",
```
Comment (JonLiu1993) [inline ports/sdformat12/vcpkg.json:5]:
```suggestion
  "homepage": "http://sdformat.org/",
  "license": "Apache-2.0",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
Tested usage successfully by `sdformat12:x64-windows` triplet.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by talregev** [comment]
I close this PR.
I submit sdformat with the new gz-ports PR:
https://github.com/microsoft/vcpkg/pull/28656

**Pull request closed without merging**